make/autoconf/flags-cflags.m4

```diff
@@ -134,6 +134,9 @@ AC_DEFUN([FLAGS_SETUP_WARNINGS],
 
       WARNINGS_ENABLE_ALL="-W3"
       DISABLED_WARNINGS="4800"
+      if test "x$TOOLCHAIN_VERSION" = x2017; then
```

Comment:
I would like a comment here explaining why we need to disable this warning, since it's caused by a faulty behavior in the compiler.